packages/block-editor/src/components/autocomplete/index.js

```diff
@@ -16,112 +15,41 @@ import { Autocomplete as OriginalAutocomplete } from '@wordpress/components';
  */
 import { withBlockEditContext } from '../block-edit/context';
 
-/*
- * Use one array instance for fallback rather than inline array literals
- * because the latter may cause rerender due to failed prop equality checks.
- */
-const completersFallback = [];
-
 /**
  * Wrap the default Autocomplete component with one that
  * supports a filter hook for customizing its list of autocompleters.
  *
- * Since there may be many Autocomplete instances at one time, this component
- * applies the filter on demand, when the component is first focused after
- * receiving a new list of completers.
- *
  * This function is exported for unit test.
  *
  * @param  {Function} Autocomplete Original component.
  * @return {Function}              Wrapped component
  */
 export function withFilteredAutocompleters( Autocomplete ) {
-	return class FilteredAutocomplete extends Component {
-		constructor() {
-			super();
-
-			this.state = { completers: completersFallback };
-
-			this.saveParentRef = this.saveParentRef.bind( this );
-			this.onFocus = this.onFocus.bind( this );
-		}
-
-		componentDidUpdate() {
-			const hasFocus = this.parentNode.contains( document.activeElement );
-
-			/*
-			 * It's possible for props to be updated when the component has focus,
-			 * so here, we ensure new completers are immediately applied while we
-			 * have the focus.
-			 *
-			 * NOTE: This may trigger another render but only when the component has focus.
-			 */
-			if ( hasFocus && this.hasStaleCompleters() ) {
-				this.updateCompletersState();
-			}
-		}
-
-		onFocus() {
-			if ( this.hasStaleCompleters() ) {
-				this.updateCompletersState();
-			}
-		}
-
-		hasStaleCompleters() {
-			return (
-				! ( 'lastFilteredCompletersProp' in this.state ) ||
-				this.state.lastFilteredCompletersProp !== this.props.completers
+	return ( props ) => {
+		let { completers = [] } = props;
+
+		if ( hasFilter( 'editor.Autocomplete.completers' ) ) {
+			completers = applyFilters(
```

Comment:
I think a `useFilter` React hook would be best here as it would ensure the filter is reapplied if a new one is added after the rendering happens (reactiveness) and it could also include any performance optim. I know this is being worked on separately though.

Reply:
Yeah, that would be great!